static/app/views/settings/projectPerformance/projectPerformance.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import {RouteComponentProps} from 'react-router';
+import styled from '@emotion/styled';
+
+import Button from 'app/components/button';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import {PanelItem} from 'app/components/panels';
+import {t} from 'app/locale';
+import {Organization, Project} from 'app/types';
+import routeTitleGen from 'app/utils/routeTitle';
+import AsyncView from 'app/views/asyncView';
+import Form from 'app/views/settings/components/forms/form';
+import JsonForm from 'app/views/settings/components/forms/jsonForm';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+
+import {Field} from '../components/forms/type';
+
+type Props = RouteComponentProps<{orgId: string; projectId: string}, {}> & {
+  organization: Organization;
+  project: Project;
+};
+
+type ProjectThreshold = {
+  id?: string;
+  threshold: string;
+  metric: string;
+  editedBy?: string;
+};
+
+type State = AsyncView['state'] & {
+  threshold: ProjectThreshold;
+};
+
+class ProjectPerformance extends AsyncView<Props, State> {
+  getTitle() {
+    const {projectId} = this.props.params;
+
+    return routeTitleGen(t('Performance'), projectId, false);
+  }
+
+  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
+    const {params} = this.props;
+    const {orgId, projectId} = params;
+
+    const endpoints: ReturnType<AsyncView['getEndpoints']> = [
+      ['threshold', `/projects/${orgId}/${projectId}/transaction-threshold/configure/`],
+    ];
+
+    return endpoints;
+  }
+
+  handleDelete = () => {
+    const {orgId, projectId} = this.props.params;
+
+    this.setState({
+      loading: true,
+    });
+
+    this.api.request(`/projects/${orgId}/${projectId}/transaction-threshold/configure/`, {
+      method: 'DELETE',
+      complete: () => this.fetchData(),
+    });
+  };
+
+  getEmptyMessage() {
+    return t('There is no threshold set for this project.');
+  }
+
+  renderLoading() {
+    return (
+      <div style={{margin: '18px 18px 0'}}>
```

Comment:
Directly styled? Maybe move this to a `styled` emotion div.

Reply:
addressed in daf94f8052238644042533691a9de8f10d8ae7a1